### ISSUE TITLE
No result using `Dataset(store="Oxigraph")`

### ISSUE THREAD

**Issue by namedgraph** [oxigraph/oxrdflib#28]
Hi,

I'm getting the expected result using `Dataset()` but not `Dataset(store="Oxigraph")`:
```python
from rdflib import Dataset

query = """
    SELECT *
    WHERE
    {{
        ?s ?p ?o .
    }}
        """

data="<http://s> <http://p> <http://o> ."

print("Dataset")
ds = Dataset()
ds.default_context.parse(data=data, format="ntriples")

qres = ds.query(query)
for row in qres:
    print(row)

print("Dataset(store=\"Oxigraph\")")
ds = Dataset(store="Oxigraph")
ds.default_context.parse(data=data, format="ntriples")

qres = ds.query(query)
for row in qres:
    print(row)
```
Output:
```
Dataset
(rdflib.term.URIRef('http://o'), rdflib.term.URIRef('http://p'), rdflib.term.URIRef('http://s'))
Dataset(store="Oxigraph")
```

**Comment by Tpt** [comment]
Hi! It's yet an other problem with rdflib graph management...
-  Rdflib sets to `self.indentifier` the default graph when calling the internal store `query`: https://github.com/RDFLib/rdflib/blob/main/rdflib/graph.py#L1553
-  Rdflib sets to `self.default_context` the default graph when running the rdflib query evaluator: https://github.com/RDFLib/rdflib/blob/main/rdflib/plugins/sparql/sparql.py#L272
Hence the behavior difference...

**Comment by namedgraph** [comment]
Does https://github.com/RDFLib/rdflib/issues/2375 address this?

**Comment by aucampia** [comment]
@namedgraph hard to say, I think it may not, the store interface is a bit weird to be quite frank, I know @gjhiggins has worked a bit on improving it. It would be nice to redesign it entirely, but that is easier said than done in some ways. 

I will add an RDFLib issue for this to see.

Just a note though, if you are not too worried about interface breakages, I think you should consider using RDFLib main branch. I'm rather strict with what goes into master, and for the most part, all new PRs should have typing, so if you do use it in some mission-critical context with something like dependabot and mypy in CI you should pick up breakages before integration.

**Comment by Tpt** [comment]
> Does https://github.com/RDFLib/rdflib/issues/2375 address this?

Sadly not. `Dataset.indentifier` and `Dataset.indentifier` are still different.

**Comment by gjhiggins** [comment]
> I'm getting the expected result using `Dataset()` but not `Dataset(store="Oxigraph")`:

Given your query, you need to set the default union to True (it's False by default):

```python
    ds = Dataset(store="Oxigraph", default_union=True)
```

**Comment by aucampia** [comment]
I made a list of some of the issues with the Store interface that I can think of, would be good to get some more feedback, I will try to collect it in a separate issue sometime so we can keep it in mind for future changes:

- https://github.com/RDFLib/rdflib/discussions/2437#discussioncomment-6166326

**Comment by namedgraph** [comment]
@aucampia can this be addressed in RDFLib 7.x? :) Not sure if there's a dedicated issue for this.

**Comment by aucampia** [comment]
> @aucampia can this be addressed in RDFLib 7.x? :) Not sure if there's a dedicated issue for this.

I made an issue: https://github.com/RDFLib/rdflib/issues/2538

And the fix for it is also described there, it's just a couple of lines, if someone adds it with some tests I will include it in the next release. I don't think it's a breaking change, so it could go into 7.x, but I'm not sure when that will be. If it is quite urgent, it can be sooner rather than later.

**Comment by aucampia** [comment]
@namedgraph I will try fix it this weekend.

**Comment by aucampia** [comment]
@Tpt if you could have a brief look at <https://github.com/RDFLib/rdflib/pull/2546> I would appreciate it.

**Comment by namedgraph** [comment]
@Tpt can you please take a look? :)

**Comment by namedgraph** [comment]
So this still doesn't work 😢

**Comment by Tpt** [comment]
This is now fixed in rdflib 7.1. I have integrated a test to check it works: https://github.com/oxigraph/oxrdflib/blob/409d3dbaba04211bb85f479b3206a59dcfa4e1e3/tests/test_sparql.py#L62
